crates/xtask-unpublished/src/xtask.rs

```diff
@@ -117,47 +122,107 @@ fn unpublished(args: &clap::ArgMatches, config: &mut cargo::util::Config) -> car
+        return Ok(());
     }
-    for (name, last, current) in results {
-        if let Some(last) = last {
-            println!("{name} {last} {current}");
-        } else {
-            println!("{name} - {current}");
+
+    output_table(results)?;
+
+    Ok(())
+}
+
+/// Outputs a markdown table of publish status for each members.
+///
+/// ```text
+/// | name                            | crates.io | local  | published? |
+/// | ----                            | --------- | -----  | ---------- |
+/// | cargo                           | 0.70.1    | 0.72.0 | no         |
+/// | cargo-credential                | 0.1.0     | 0.2.0  | no         |
+/// | cargo-credential-1password      | 0.1.0     | 0.2.0  | no         |
+/// | cargo-credential-gnome-secret   | 0.1.0     | 0.2.0  | no         |
+/// | cargo-credential-macos-keychain | 0.1.0     | 0.2.0  | no         |
+/// | cargo-credential-wincred        | 0.1.0     | 0.2.0  | no         |
+/// | cargo-platform                  | 0.1.2     | 0.1.3  | no         |
+/// | cargo-util                      | 0.2.3     | 0.2.4  | no         |
+/// | crates-io                       | 0.36.0    | 0.36.1 | no         |
+/// | home                            | 0.5.5     | 0.5.6  | no         |
+/// ```
+fn output_table(results: Vec<Record>) -> fmt::Result {
+    let mut results: Vec<_> = results
+        .into_iter()
+        .map(|e| {
+            (
+                e.0,
+                e.1.unwrap_or("-".to_owned()),
+                e.2,
+                if e.3 { "yes" } else { "no" }.to_owned(),
+            )
+        })
+        .collect();
+
+    let header = (
+        "name".to_owned(),
+        "crates.io".to_owned(),
+        "local".to_owned(),
+        "published?".to_owned(),
+    );
+    let separators = (
+        "-".repeat(header.0.len()),
+        "-".repeat(header.1.len()),
+        "-".repeat(header.2.len()),
+        "-".repeat(header.3.len()),
+    );
+    results.insert(0, header);
+    results.insert(1, separators);
+
+    let max_col_widths = results
+        .iter()
+        .map(|(name, last, local, bump)| (name.len(), last.len(), local.len(), bump.len()))
+        .reduce(|(c0, c1, c2, c3), (f0, f1, f2, f3)| {
+            (c0.max(f0), c1.max(f1), c2.max(f2), c3.max(f3))
+        })
+        .unwrap();
+
+    let print_space = |out: &mut dyn Write, n| {
+        for _ in 0..(n + 1) {
+            write!(out, " ")?;
         }
+        fmt::Result::Ok(())
+    };
+
+    let out = &mut String::new();
+    for (name, last, local, bump) in results {
+        write!(out, "| {name}")?;
+        print_space(out, max_col_widths.0 - name.len())?;
+
+        write!(out, "| {last}")?;
+        print_space(out, max_col_widths.1 - last.len())?;
+
+        write!(out, "| {local}")?;
+        print_space(out, max_col_widths.2 - local.len())?;
+
+        write!(out, "| {bump}")?;
+        print_space(out, max_col_widths.3 - bump.len())?;
+
+        writeln!(out, "|")?;
```

Comment:
Instead or `print_space` you can do
```rust
let field = name;
let padding = max_col_widths.0;
write!(out, "| {field:padding$}")?;
```

See https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8afa23bc072b2fb41843b0c29e6ebfc7